networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/gossip/GossipMessageHandler.java

```diff
@@ -69,7 +84,35 @@ static PubsubPublisherApi createPublisher(final Gossip gossip, final PrivKey pri
       final EventBus eventBus,
       final ChainStorageClient chainStorageClient) {
     return List.of(
-        new AttestationTopicHandler(publisher, eventBus, chainStorageClient),
+        new AggregateTopicHandler(publisher, eventBus, chainStorageClient),
         new BlocksTopicHandler(publisher, eventBus, chainStorageClient));
   }
+
+  @Subscribe
+  public void registerAttestationTopicHandlers(CommitteeAssignmentEvent assignmentEvent) {
+    List<Integer> committeeIndices = assignmentEvent.getCommitteeIndices();
+    for (int i : committeeIndices) {
+      if (!attestationTopicHandlers.containsKey(i)) {
+        AttestationTopicHandler topicHandler =
+            new AttestationTopicHandler(publisher, eventBus, chainStorageClient, i);
+        attestationTopicHandlers.put(i, topicHandler);
```

Comment:
Since we use an `AsyncEventBus` events will arrive on different threads so we'll need to make this thread safe.  Given that these events only happen once an epoch or so, I'd suggest just making the two `@Subscribe` methods `synchronized` and keeps things simple.

Reply:
makes sense. done.